### ISSUE TITLE
Session Lost After 1 hour Idle Time

### ISSUE THREAD

**Issue by asadalarma** [strangerstudios/paid-memberships-pro#2184]
**Describe the bug**
A clear and concise description of what the bug is. Please be as descriptive as possible; issues lacking detail, or for any other reason than to report a bug, may be closed or left unattended.

I am using Paid Membership Pro version 2.7.5 and using paypal express checkout method gateway . When the user tries to complete the registration , he will redirect to the sandbox or paypal page ( we ideally stop activity and do nothing ) after 1 hour we logged in as a paypal user and pay the amount , the user is redirected to the checkout page and gets the error message 

**we have sandbox paypal screen for more than 1 hr and then complete the process.and gets the error message**

 Order status is success but user id that needs to be attached on orders table coming none and no record on membership table

**Screenshots**
![Screenshot from 2022-08-03 19-15-15](https://user-images.githubusercontent.com/41333412/182632755-f89b67e1-9fa6-4f64-b52b-9a21df6720ba.png)

![image (1)](https://user-images.githubusercontent.com/41333412/182632815-a1d1715b-fc6b-4780-9308-cdcbc4339361.png)

**Expected behavior**
User should go the confirmation page when he paid and process should complete.

**WordPress Environment**
<details>
```
Wordpress latest version , PHP 7.4 and have pmpro register helper  plugin
```
</details>

**Comment by asadalarma** [comment]
Do the upgrading the version to latest or paid version will solve problem or not !

**Comment by mircobabini** [comment]
> Do the upgrading the version to latest or paid version will solve problem or not !

Nope. You can try this workaround though: https://gist.github.com/mircobabini/13c083d7a2b312848f97496126d2aff2

Let us know if you test it out.

**Comment by asadalarma** [comment]
ok thanks let me check and acknowledge you microbabini i was thinking the issue fix was 
PPE: failed account creation when session is lost #1338

**Comment by mircobabini** [comment]
That issue is a full report of the investigation. Still useful but not really a solution.

**Comment by asadalarma** [comment]
mircobabini also need help in variable pricing

**Comment by asadalarma** [comment]
I have used this filter
 add_filter('pmpro_checkout_level','pmpro_checkout_level');


function pmpro_checkout_level($level){
    if(!is_free_membership($level)){
        $uuid = !empty($_COOKIE["uuid"]) ? $_COOKIE["uuid"] : (!empty($_SESSION['uuid']) ? $_SESSION['uuid'] : null);
        $payment_amount = !empty(function_exists('get_amount_from_uuid')) ?  get_amount_from_uuid($uuid) : null;
        $level->initial_payment = number_format($payment_amount, 2);    
    }
    //$level->initial_payment = '25.00';
    return $level;
}



/*
 * Check the membership is free or not
 */

function is_free_membership($level){
    return (number_format($level->initial_payment) == 0 ) && (number_format($level->billing_amount) == 0 )  && ( number_format($level->billing_amount) == 0 );
}

**Comment by mircobabini** [comment]
I'm sorry but this is not the right place to get general support unrelated to a bug. Feel free to reach us on Slack or open a support ticket on site.
